build.sh

```diff
@@ -440,6 +442,44 @@ case "$COMMAND" in
             | sed 's/.*/-framework &/' \
             | xargs xcodebuild -create-xcframework -allow-internal-distribution -output build/RealmSwift.xcframework
 
+        # Because we have a module named Realm and a type named Realm we need to manually resolve the naming
+        # collisions that are happening. These collisions create a red herring which tells the user the xcframework
+        # was compiled with an older Swift version and is not compatible with the current compiler.
+        count=$(find build/RealmSwift.xcframework -name "*.swiftinterface" | wc -l)
+        if [ $count > 0 ]
+        then 
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/Realm\.//g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/import Private/import Realm.Private/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/RealmSwift.Configuration/RealmSwift.Realm.Configuration/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/extension Configuration/extension Realm.Configuration/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/RealmSwift.Error/RealmSwift.Realm.Error/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/extension Error/extension Realm.Error/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/RealmSwift.AsyncOpenTask/RealmSwift.Realm.AsyncOpenTask/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/RealmSwift.UpdatePolicy/RealmSwift.Realm.UpdatePolicy/g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/RealmSwift.Notification /RealmSwift.Realm.Notification /g' {} \;
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/RealmSwift.Notification,/RealmSwift.Realm.Notification,/g' {} \;
+            # Generics will use τ_1_0 which needs to be changed to the correct type name.
+            find build/RealmSwift.xcframework -name "*.swiftinterface" -exec sed -i '' 's/τ_1_0/V/g' {} \;
+        fi 
+
+        exit 0
+        ;;
+
+    "verify-xcframework-evolution-mode")
+        export REALM_EXTRA_BUILD_ARGUMENTS="$REALM_EXTRA_BUILD_ARGUMENTS REALM_BUILD_LIBRARY_FOR_DISTRIBUTION=YES"
+        # set the Xcode version to the oldest
+        export REALM_XCODE_VERSION=$REALM_XCODE_OLDEST_VERSION
+        export REALM_SWIFT_VERSION=$REALM_SWIFT_OLDEST_VERSION
```

Comment:
Setting both REALM_SWIFT_VERSION and REALM_XCODE_VERSION is unnecessary. The Swift version is automatically set from the Xcode version, and it being a separate thing is just old cruft from when Xcode packaged multiple versions of the Swift toolchain.